esmvaltool/diag_scripts/fire/diagnostic_run_confire.py

```diff
@@ -0,0 +1,1230 @@
+
+
+# /libs/iris_plus.py
+def _sort_time(
+    cube: iris.cube.Cube, field: str, filename: str
+) -> iris.cube.Cube:
+    """Sort time dimension in the iris cube.
+
+    Parameters
+    ----------
+        cube: iris cube
+            Input cube.
+        field: str
+            Variable name in cube.
+        filename: str
+            Filename of cube.
+
+    Returns
+    -------
+        cube: iris cube
+            Cube with sorted and added time dimensions.
+    """
+    logger.debug("Sorting time for variable %s in cube %s", field, filename)
+
+    cube.coord("time").bounds = None
+    tcoord = cube.coord("time")
+    tcoord.units = cf_units.Unit(tcoord.units.origin, calendar="gregorian")
+    tcoord.convert_units("days since 1661-01-01 00:00:00")
+    tcoord.units = cf_units.Unit(
+        tcoord.units.origin, calendar="proleptic_gregorian"
+    )
+    cube.remove_coord("time")
+    cube.add_dim_coord(tcoord, 0)
+
+    try:
+        iris.coord_categorisation.add_year(cube, "time")
+    except (ValueError, iris.exceptions.CannotAddError) as expt:
+        logger.debug("_sort_time could not add year %s", expt)
+
+    try:
+        try:
+            cube.remove_coord("month")
+        except (ValueError, iris.exceptions.CoordinateNotFoundError) as expt:
+            logger.debug("_sort_time could not remove month %s", expt)
+        iris.coord_categorisation.add_month_number(cube, "time", name="month")
+    except (ValueError, iris.exceptions.CannotAddError) as expt:
+        logger.debug("_sort_time could not add month number %s", expt)
+
+    try:
+        del cube.attributes["history"]
+    except (ValueError, iris.exceptions.IrisError) as expt:
+        logger.debug("_sort_time could not remove cube history %s", expt)
+
+    return cube
+
+
+def _insert_data_into_cube(
+    data: np.array, eg_cube: iris.cube.Cube, mask: np.array | None = None
+) -> iris.cube.Cube:
+    """Insert data into cube following mask.
+
+    Parameters
+    ----------
+        data: np.array
+            data that we want to insert into the cube.
+            Should have same shape of eg_cube, or same length as eg_cube
+            or length equal to Trues in mask.
+        eg_cube: iris cube
+            The cube we want to insert data into.
+        mask: Boolean array
+            Array of shape or length x where True, will inster data.
+            Default of None which means True for all points in eg_cube.
+
+    Returns
+    -------
+        pred_cube: iris cube
+            cube with data replaced by x
+    """
+    pred_cube = eg_cube.copy()
+    # Only keep parent's attributes
+    attrs_parent = {
+        key: item
+        for key, item in eg_cube.attributes.items()
+        if key[:6] == "parent"
+    }
+    pred_cube.attributes = attrs_parent
+    # Copy data
+    pred = pred_cube.data.copy().flatten()
+
+    if mask is None:
+        pred[:] = data
+    else:
+        pred[mask] = data
+
+    pred_cube.data = pred.reshape(pred_cube.data.shape)
```

Comment:
prob a good idea to keep things lazy and use `core_data` instead of `data` - unless `reshape` is called at the very beginning of the stack and a compute then is needed, in that case things get blown up in memory anyway